File: ProjetAttractionJPA/src/main/java/metier/Commodite.java
package metier;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Commodite extends Element {

	private String nom;
	@Column(name="prix_acquisition")
	private double prixAcquisition;
	private double taille;
	
	
	public Commodite(int id, String nom, double prixAcquisition, double taille) {
		super(id);
		this.nom = nom;
		this.prixAcquisition = prixAcquisition;
		this.taille = taille;
	}
	
	
	
	public Commodite(String nom, double prixAcquisition, double taille) {
		super();
		this.nom = nom;
		this.prixAcquisition = prixAcquisition;
		this.taille = taille;
	}
	
	public Commodite(){}
	
	
	public String getNom() {
		return nom;
	}
	public void setNom(String nom) {
		this.nom = nom;
	}
	
	public double getPrixAcquisition() {
		return prixAcquisition;
	}
	public void setPrixAcquisition(double prixAcquisition) {
		this.prixAcquisition = prixAcquisition;
	}
	
	public double getTaille() {
		return taille;
	}
	public void setTaille(double taille) {
		this.taille = taille;
	}
	
	
	@Override
	public String toString() {
		return "La commodité" + nom + " (id--> " + id+")\n"+"\t|Taille: " + taille + " m²"+"\t|Prix d'acquisition:" + prixAcquisition + "€\n";
	}
	
	
	
}

File: ProjetAttractionAngular/src/app/model/commodite.ts
import { Element } from './element';

export class Commodite extends Element{

  constructor(
    private _nom: string,
  	private _prixAcquisition: number,
  	private _taille: number,
    id?: number
  ){super();}


	public get nom(): string {
		return this._nom;
	}

	public set nom(nom: string) {
		this._nom = nom;
	}

	public get prixAcquisition(): number {
		return this._prixAcquisition;
	}

	public set prixAcquisition(prixAcquisition: number) {
		this._prixAcquisition = prixAcquisition;
	}

	public get taille(): number {
		return this._taille;
	}

	public set taille(taille: number) {
		this._taille = taille;
	}
}
